Port URLSchemeTaskParameters to the new IPC serialization format

cdumez opened the pull request:
#### 0e86eeb847bf05a1216b1944dac816acd3fe05fb
<pre>
Port URLSchemeTaskParameters to the new IPC serialization format
<a href="https://bugs.webkit.org/show_bug.cgi?id=264769">https://bugs.webkit.org/show_bug.cgi?id=264769</a>

Reviewed by Alex Christensen.

* Source/WebKit/CMakeLists.txt:
* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/Shared/URLSchemeTaskParameters.cpp:
(WebKit::URLSchemeTaskParameters::URLSchemeTaskParameters):
(WebKit::URLSchemeTaskParameters::requestBody const):
(WebKit::URLSchemeTaskParameters::encode const): Deleted.
(WebKit::URLSchemeTaskParameters::decode): Deleted.
* Source/WebKit/Shared/URLSchemeTaskParameters.h:
(WebKit::URLSchemeTaskParameters::URLSchemeTaskParameters):
* Source/WebKit/Shared/URLSchemeTaskParameters.serialization.in: Added.
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Canonical link: <a href="https://commits.webkit.org/270689@main">https://commits.webkit.org/270689@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/f91b366eb2ab48d754d25f05e6681f69230fdb5e

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/26087 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/4697 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/27361 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/28183 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/23888 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/6462 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/2112 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/23936 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/26342 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/3558 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/22461 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/28759 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/3172 "Passed tests") | | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/29468 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/23791 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/23795 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/27353 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/3210 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/1409 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/4611 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/3674 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/3361 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/3531 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

achristensen07 reviewed (approved):
I think the time may have come where we always include the HTTP body when serializing a ResourceRequest.

cdumez committed:
Port URLSchemeTaskParameters to the new IPC serialization format
https://bugs.webkit.org/show_bug.cgi?id=264769

Reviewed by Alex Christensen.

* Source/WebKit/CMakeLists.txt:
* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/Shared/URLSchemeTaskParameters.cpp:
(WebKit::URLSchemeTaskParameters::URLSchemeTaskParameters):
(WebKit::URLSchemeTaskParameters::requestBody const):
(WebKit::URLSchemeTaskParameters::encode const): Deleted.
(WebKit::URLSchemeTaskParameters::decode): Deleted.
* Source/WebKit/Shared/URLSchemeTaskParameters.h:
(WebKit::URLSchemeTaskParameters::URLSchemeTaskParameters):
* Source/WebKit/Shared/URLSchemeTaskParameters.serialization.in: Added.
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Canonical link: https://commits.webkit.org/270689@main
